include/mo.php

```diff
@@ -74,4 +74,24 @@ public function import_from_db( $lang ) {
 	public function delete_entry( $string ) {
 		unset( $this->entries[ $string ] );
 	}
+
+	/**
+	 * Translates a string or returns false if the translation is not found.
+	 * Contrary to `self::translate()`, this method doesn't fallback to the source string.
+	 *
+	 * @since 3.7
+	 *
+	 * @param string $source The source string to translate.
+	 * @return string|false The translated string or false if not found.
+	 */
+	public function translate_or_fail( string $source ) {
+		$entry = new Translation_Entry( array( 'singular' => $source ) );
+		$entry = $this->translate_entry( $entry );
+
+		if ( ! $entry instanceof Translation_Entry || empty( $entry->translations ) ) {
+			return false;
```

Comment:
I would rather always return a string.
```suggestion
			return '';
```

Reply:
Done in 813cda54.